### DISCUSSION TITLE
Can two surveyors use qfield on the same project and sync project so all data is together

### DISCUSSION THREAD

**Discussion by AVRIO-Environmental** [opengisch/QField#4234] [Q&A]
I am currently working on a project with another team member and we both need to survey different areas within the project at the same time. If we both work on the project can we both upload our changes onto QField Cloud and merge our changes? Any advice I could get on this would be really appreciated.

**Comment by ces-lop10** [comment]
En mi experiencia, es importante tener en cuenta que las cuentas "premium" son individuales, con base en esto, si puedes trabajar a la par dos personas en el mismo proyecto pero necesitarían tener una cuenta cada quien.

**Comment by Meykaefer** [comment]
Hi Avrio,

It should be possible, but it would depend on the project setup. First off, are you having a private or public project?

If it is a private project, one team member would require the pro subscription to invite a collaborator, or both team members could be members of the same organisation.

The second condition is how the project is set up. If the feature ID column is the standard 'fid' setting, you might overite each other's changes or get an error. Using a different feature ID column, like a UUID generator, should prevent this and enable the merging of simultaneously collected data. I had no chance to test this by myself so better do a test run. It would be interesting to know if it works.

**Reply by BoswachterMarc** [reply]
Offline syncing (over cable) with QFieldSync might be a solution if the 2 persons meet regularly? See https://docs.qfield.org/get-started/tutorials/get-started-qfs/

![image](https://github.com/opengisch/QField/assets/57868975/a4cccc80-0dc9-4288-83ff-df0792833bf6)

**Reply by Meykaefer** [reply]
This could work as well.

I think it comes down to the project setup due to the nature of collecting data simultaneously.

At the very bottom of the tutorial, it says the following:

> Make sure that you synchronize your data back only once. That means, if you go out again to collect more data, you should create a new QField package before to avoid sync problems later on (e.g.  duplicates). 

I don't know how well this analogy works, but imagine your .gpkg or .shp as a folder on your PC with the feature ID column being the file name. When you collect data, you automatically assign a fid (usually an INT): 1, 2, 3, 4,...
A second team member collecting data simultaneously will assign identical fid's (when they are INT); 1, 2, 3, 4,...

If you now try to bring both data sets together, you should encounter a similar scenario as if you attempted copying files into a folder where already files with the same name exist, and you will get an error message telling you that the file already exists. It will ask you what to do (overwrite, rename, do nothing). I don't know what QGIS would offer the user in this case. It could either fail or data could be overwritten and lost. 

And this also highlights another thing to consider. If photos are collected as part of the project and labelled by a custom expression (i.e. photo_"number".jpg), both team members could have photos labelled "photo_1.jpg" in their DCIM folder, which could cause problems when trying to get the data into one location.
